doc/decisions/library_directory_structure.md

```diff
@@ -2,21 +2,76 @@
 
 ## Problem
 
-Currently src/libs/elektra contains the source of several libraries. Each of the libraries should have its own folder.
+Currently `src/libs/elektra` contains the source of several libraries. Each of the libraries should have its own folder.
 Furthermore, we need to allow rust source to exist in parallel.
 
 ## Constraints
 
+- It must be possible to set up include-paths via CMake so that `#include "header.h"` works the same in the build directory and when installed.
+
 ## Assumptions
 
 ## Considered Alternatives
 
 ## Decision
 
+We use this directory tree (YAML syntax for highlighting):
+
+```yaml
+doc:
+examples:
+scripts:
+src:
+  include:
+   # headers that don't belong to a specific library
+  libs:
+    core-c: C implementation of libelektra-core.so
+    core-rust: Rust implementation of libelektra-core.so
+    opts: libelektra-opts.so has only a single implementation, so no suffix
+    mount-c: mounting library written in C
+  # plugins and tools use the same suffix idea as above
+  plugins:
+  tools:
+  bindings:
+    rust: Rust bindings
+    cpp: C++ bindings
+    java: Java bindings
+  tests:
+    shell: shell script tests
+    cframework: framework for C tests
+    gtest-framework: framework for C++ tests
+    abi: ABI compatibility tests
+  # build tool configuration files to create a single root project for every language
+  # this is improves IDE support
+  CMakeLists.txt
+  build.gradle
+  Cargo.toml
+```
```

Comment:
@markus2330 @lawli3t I have now updated this document with the results from https://github.com/ElektraInitiative/libelektra/pull/4223#discussion_r809951795.